src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -1279,10 +1279,11 @@ internal static void CalculateParentAssemblyDirectories(List<string> parentRefer
             // First, look for the dependency in the parents' directories. Unless they are resolved from the GAC or assemblyFoldersEx then
             // we should make sure we use the GAC and assemblyFolders resolvers themserves rather than a directory resolver to find the reference.
             // This way we dont get assemblies pulled from the GAC or AssemblyFolders but dont have the marking that they were pulled form there.
-            var parentReferenceFolders = new List<string>();
+            var parentReferenceDirectoriesMap = new Dictionary<string, List<string>>();
             foreach (Reference parentReference in reference.GetDependees())
             {
-                CalculateParentAssemblyDirectories(parentReferenceFolders, parentReference);
+                parentReferenceDirectoriesMap[parentReference.FullPath] = new List<string>();
```

Comment:
There could be more than one, right? I think this needs to be a get-or-add operation.

Reply:
I'm sorry, I didn't quite understand what you meant. Could you please rephrase or explain it a bit more?

Reply:
Yes, you are right. It is not guaranteed. I changed this part of the code. Thanks!